Trips index page

lxingern opened the pull request:
Created an index page for the user's trips, which is almost the same as the index page for trails, but I implemented two tabs for upcoming and past trips which the user can switch between.

I used Stimulus for this, so as usual, remember to run 'yarn build --watch'.

barbayjuliette commented:
- When I create a trip, I have multiples of 5. But If I click now at 14.48 to create a trip, I have multiple of 5 from 48, so if I click on 'plus', the minutes show 53. Should not show 48 but next multiple of 5 (50)
- Past trip: the date showed on top is unformatted. 
<img width="537" alt="Screenshot 2022-11-27 at 14 40 31" src="https://user-images.githubusercontent.com/109951226/204122807-6244bd1c-54bd-4e38-8606-85e334e43307.png">

barbayjuliette commented on Gemfile:
```diff
@@ -41,16 +41,13 @@ gem "pg_search"
 gem "tzinfo-data", platforms: %i[ mingw mswin x64_mingw jruby ]
 
 gem "cloudinary"
```
Why did you delete those gems?

lxingern replied:
When I was running bundle install, there was a warning that some gems were listed more than once which may create problems in the future, so I just deleted the duplicates.

barbayjuliette commented on app/views/trips/index.html.erb:
```diff
@@ -2,46 +2,101 @@
+    </div>
+  </div>
 
-      <%# <div style="height: 200px; width: 300px; background-position: center; background-repeat: no-repeat; background-size: cover; background-image: url('https://lonelyplanetimages.imgix.net/a/g/hi/t/a2aa48e66952bf8816898991072e32f5-macritchie-reservoir.jpg')">
```
Would it be possible to use the cards partial? (Use partial but just add trip date on top)
--> I think the easiest way would be to just add the date (if that's the only difference) inside the partial, with a condition that the trip date only appears if it has a date in the future, or something like that (but then in the index, you would still see the date for upcoming trips)
So that the changes we use for the cards are reflected here, since I have changed the cards.

lxingern commented:
- Formatted the dates on the cards for past trips
- Changed styling of the submit button in the date select pop up when the user creates a trip to be consistent with other buttons
- Updated styling of cards in the trips index to match the trails index

Will try to refactor and use partials at some point... it will take some time to incorporate the partial in this page since the instance variable is different. We would have to change the variable in the partial code and update the <%= render %> lines in other views to specify the locals

barbayjuliette commented:
When I tested before the change, I created at time = now, and then a minute later, it would be already in past trips. Right now, if I do the same, it is still in the upcoming trips. So I cannot test. Did you change the logic to only base it on the date only, not time?
